src/main/scala/com/modelfabric/sparql/stream/client/Builder.scala

```diff
@@ -244,8 +249,10 @@ object Builder {
 
   }
 
+
   private def responseToSparqlResponse(response: (Try[HttpResponse], SparqlRequest)): SparqlResponse = response match {
     case (Success(HttpResponse(StatusCodes.OK, _, _, _)), request) =>
+    // JC: can we also extract ResultSet here, instead of broadcast then zip?
```

Comment:
no, see my previous comment about the response entity parsing (unmarshalling) being async